### DISCUSSION TITLE
GC HeapHardLimit functionality within .NET Framework 4.6.2

### DISCUSSION THREAD

**Discussion by denis-dbm** [microsoft/dotnet-framework-docker#1097] [Q&A] [answered]
Hi,

I've found this [great announcement](https://github.com/microsoft/dotnet-framework-docker/issues/863) that tells about the support of _GC HeapHardLimit_ on .NET Framework 4.8. I've tried (and I was trolled :trollface: ) to test the functionality on .NET 4.6.2 (`<TargetFrameworks>`), but I checked again my tests, and it was executed on .NET Framework 4.8 in my computer (`RuntimeInformation.FrameworkDescription = ".NET Framework 4.8.9181.0"`).

I found [this change too](https://github.com/microsoft/dotnet-framework-docker/pull/842/files#diff-2463cbe89b662ac55920d48785e550b9d144d7ee73cb979fbe668b64651398c4), and I've checked the functionality, actually, doesn't run on .NET 4.8 or lower. Questions:

* Am I right? Doesn't it run on 4.6.2 or other versions lower than .NET Framewoek 4.8.x?
* What is the exact version (build number) of .NET Framework 4.8 the GC HeapHardLimit works? Should it be 4.8.1 and higher?
* Note: the application (and my tests) doesn't run in containers (yet), it just runs on Windows Server Core

Thanks!

**Comment by denis-dbm** [comment]
@mthalman / @MichaelSimons could you help me?

**Reply by MichaelSimons** [reply]
CC @Maoni0

**Reply by Maoni0** [reply]
@mangod9 has the info of the exact 4.8 version.

**Reply by mangod9** [reply, marked as answer]
yeah it should be in 4.8.1. Is it not working for you?

**Reply by denis-dbm** [reply]
@mangod9 thank you for your answer. I am confirming the minimum version required for this feature because I have sensitive scenario (a critical application; and to confirm my tests too, that is, now I see it was on 4.8.1), and [the docs](https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector) are unclear about the support of the feature on .NET Framework. Could I open an issue or a PR to state that the feature (`COMPlus_GCHeapHardLimit` and `COMPlus_GCHeapHardLimitPercent`) is supported on .NET Framework 4.8.1? If yes, to complement: does 4.8.1 support [Per-object-heap limits](https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector#per-object-heap-limits)?

Thank you!

**Reply by Maoni0** [reply]
have you confirmed you have the feature or are you still trying to confirm that? 

I meant to ask the exact version of clr.dll, something like this

`File version:     4.8.9181.0`

(this is just an example, I don't actually know the exact min version that contains heap hard limit support)

but short of that, you could get your process under a debugger and see if you can see this symbol at all - if you can, it means you are using a version with this backported -

```
0:466> x clr!*heap_hard_limit
00007ffd`098487c8 clr!SVR::gc_heap::heap_hard_limit = 0
00007ffd`0984a520 clr!WKS::gc_heap::heap_hard_limit = 0
```

this is from windbg.

if you do see this symbol, do you see it having a non zero value?

**Reply by denis-dbm** [reply]
@Maoni0 sorry if I was unclear in my last post. I confirmed it because @mangod9 told me the minimum required version (4.8.1). I only justified why I asked about the support of GC Hard Limit.
>I am confirming the minimum version required for this feature because I have sensitive scenario (a critical application; and to confirm my tests too, that is, now I see it was on 4.8.1),

`The exact version of clr.dll: 4.8.9181.0`
`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\Release: 533325`

According to above information, I have the .NET Framework 4.8.1 in my computer (not production, it's development machine), and the GC Heap Hard Limit works well as expected.

In short, GC Heap Hard Limit works only on .NET Framework 4.8.1. Unfortunately, it doesn't work for older versions (i.e. 4.6.2).

And finally:
>Could I open an issue or a PR to state that the feature (COMPlus_GCHeapHardLimit and COMPlus_GCHeapHardLimitPercent) is supported on .NET Framework 4.8.1? If yes, to complement: does 4.8.1 support [Per-object-heap limits](https://learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector#per-object-heap-limits)?

Does it make sense for you (@Maoni0 / @mangod9)? I reported this at https://github.com/dotnet/docs/issues/38052

Thank you!

**Reply by Maoni0** [reply]
sure, please feel free to submit a doc PR!

per object heap limits are not supported on .NET Framework.
